label_maker/utils.py

```diff
@@ -137,15 +138,29 @@ def get_tile_tif(tile, imagery, folder, kwargs):
         window = ((top, bottom), (left, right))
 
         # read the first three bands (assumed RGB) of the TIF into an array
-        data = np.empty(shape=(3, 256, 256)).astype(src.profile['dtype'])
-        for k in (1, 2, 3):
-            src.read(k, window=window, out=data[k - 1], boundless=True)
-
+        try:
+            kwargs['band_indicies']
+            band_indicies =  tuple(kwargs['band_indicies'])
+            band_count = len(kwargs['band_indicies'])
+        except KeyError:
+            band_indicies = (1, 2, 3)
+            band_count = 3
+
+        arr_shape = tuple([band_count, 256, 256])
+        data = np.empty(shape=(arr_shape)).astype(src.profile['dtype'])
+
+        for i, k in enumerate(band_indicies):
+            src.read(k, window=window, out=data[i], boundless=True)
         # save
-        tile_img = op.join(folder, '{}{}'.format(tile, '.jpg'))
-        img = Image.fromarray(np.moveaxis(data, 0, -1), mode='RGB')
-        img.save(tile_img)
-
+        if band_indicies == (1, 2, 3):
+            tile_img = op.join(folder, '{}{}'.format(tile, '.jpg'))
+            img = Image.fromarray(np.moveaxis(data, 0, -1))
+            img.save(tile_img)
+        else:
+            tile_img = op.join(folder, '{}{}'.format(tile, '.png'))
+            with rasterio.open(tile_img, 'w', driver='png', height=256,
+                    width=256, count=band_count, dtype=profile['dtype']) as w:
+                    w.write(data)
```

Comment:
if we have more than 4 bands, we'll need to write out as (Geo)TIFF

Reply:
remove if check because everything will be saved out as tif